vault_jwt_auth_backend_role causes terraform to panic when no policies list given

Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
v0.11.11

### Affected Resource(s)
Please list the resources as a list, for example:
- vault_jwt_auth_backend_role

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "vault_jwt_auth_backend_role" "canary_jwt_auth_role" {
  backend   = "${vault_jwt_auth_backend.canary_jwt_auth.path}"
  role_name = "sl-iam"

  bound_audiences = ["StrlnAuthAud"]
  user_claim      = "sub"
  groups_claim    = "strln:groups"
  ttl             = "43200"
  max_ttl         = "43200"
}
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

### Panic Output
```
panic: interface conversion: interface {} is nil, not []interface {}
2019-02-05T11:12:06.813-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4:
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: goroutine 972 [running]:
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: github.com/terraform-providers/terraform-provider-vault/vault.jwtAuthBackendRoleRead(0xc0001f48c0, 0x19150a0, 0xc0002329a0, 0x0, 0x0)
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vault/resource_jwt_auth_backend_role.go:178 +0x15c9
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: github.com/terraform-providers/terraform-provider-vault/vault.jwtAuthBackendRoleCreate(0xc0001f48c0, 0x19150a0, 0xc0002329a0, 0x24, 0x20dd200)
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vault/resource_jwt_auth_backend_role.go:141 +0x44f
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc0005df8c0, 0xc000402370, 0xc0002c9d00, 0x19150a0, 0xc0002329a0, 0x100b501, 0xc000306b80, 0x10bab8c)
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:193 +0x358
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc000232000, 0xc0004022d0, 0xc000402370, 0xc0002c9d00, 0xc0001e5400, 0x10f3fcc, 0x231ab00)
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:259 +0x9c
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc00065b400, 0xc0002c9ca0, 0xc00036eca0, 0x0, 0x0)
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:488 +0x57
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: reflect.Value.call(0xc0003464e0, 0xc000112800, 0x13, 0x192bb09, 0x4, 0xc000306f18, 0x3, 0x3, 0xc00038aa40, 0x1012ea7, ...)
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: 	/opt/goenv/versions/1.11.0/src/reflect/value.go:447 +0x449
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: reflect.Value.Call(0xc0003464e0, 0xc000112800, 0x13, 0xc000426f18, 0x3, 0x3, 0x12a05f200, 0xc000426f10, 0xc000426fb8)
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: 	/opt/goenv/versions/1.11.0/src/reflect/value.go:308 +0xa4
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: net/rpc.(*service).call(0xc0004ad100, 0xc00034e190, 0xc0002ee370, 0xc0002ee380, 0xc000170700, 0xc0003ec740, 0x17b8280, 0xc0002c9ca0, 0x16, 0x17b82c0, ...)
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: 	/opt/goenv/versions/1.11.0/src/net/rpc/server.go:384 +0x14e
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: created by net/rpc.(*Server).ServeCodec
2019-02-05T11:12:06.814-0800 [DEBUG] plugin.terraform-provider-vault_v1.5.0_x4: 	/opt/goenv/versions/1.11.0/src/net/rpc/server.go:481 +0x47e
```

### Expected Behavior
Either it should fail saying that a "policies" list is required (lazy solution), or it should recognize that no "policies" attribute was given so it is nil, not a slice. This can be done via type assertion.
ie
```
switch policiesListInterface.(type) {
    case nil: 
        // no policies given
    case []interface{}:
        // policies were given
}
```

### Actual Behavior
panic because of a missing type assertion

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-1234

--- a/vault/resource_jwt_auth_backend_role.go
+++ b/vault/resource_jwt_auth_backend_role.go
@@ -188,7 +188,10 @@ func jwtAuthBackendRoleRead(d *schema.ResourceData, meta interface{}) error {
 
 	d.Set("user_claim", resp.Data["user_claim"].(string))
 
-	policies := util.JsonStringArrayToStringArray(resp.Data["policies"].([]interface{}))
+	policies := make([]string, 0)
+	if resp.Data["policies"] != nil {
+		policies = util.JsonStringArrayToStringArray(resp.Data["policies"].([]interface{}))
+	}
 	err = d.Set("policies", policies)
 	if err != nil {
 		return fmt.Errorf("error setting policies in state: %s", err)

--- a/vault/resource_jwt_auth_backend_role_test.go
+++ b/vault/resource_jwt_auth_backend_role_test.go
@@ -85,13 +85,7 @@ func TestAccJWTAuthBackendRole_basic(t *testing.T) {
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
 						"role_name", role),
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"policies.#", "3"),
-					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"policies.1971754988", "default"),
-					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"policies.232240223", "prod"),
-					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"policies.326271447", "dev"),
+						"policies.#", "0"),
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
 						"ttl", "0"),
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
@@ -131,13 +125,7 @@ func TestAccJWTAuthBackendRole_update(t *testing.T) {
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
 						"role_name", role),
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"policies.#", "3"),
-					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"policies.1971754988", "default"),
-					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"policies.232240223", "prod"),
-					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
-						"policies.326271447", "dev"),
+						"policies.#", "0"),
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
 						"ttl", "0"),
 					resource.TestCheckResourceAttr("vault_jwt_auth_backend_role.role",
@@ -368,7 +356,6 @@ resource "vault_jwt_auth_backend_role" "role" {
 
   bound_audiences = ["https://myco.test"]
   user_claim = "https://vault/user"
-  policies = ["default", "dev", "prod"]
 }`, backend, role)
 }
 

